src/frontend/packages/lib_components/src/common/queries/fetchResponseHandler.ts

```diff
@@ -0,0 +1,48 @@
+import { FetchResponseError } from 'utils/errors/exception';
+
+interface fetchResponseHandlerOptions {
+  errorMessage?: string | { [key: number]: string }; // 'Bad Request' | { 400: 'Bad Request', 401: 'Unauthorized' }
+  invalidStatus?: number[];
+  withoutBody?: boolean;
+}
+
+export const fetchResponseHandler = async <T>(
+  response: Response,
+  options?: fetchResponseHandlerOptions,
+): Promise<T> => {
+  const { errorMessage, invalidStatus, withoutBody } = { ...options };
+
+  if (response.ok) {
+    if (withoutBody) {
+      return {} as unknown as T;
+    } else {
+      return (await response.json()) as T;
+    }
+  }
+
+  let bodyResponse;
+  try {
+    bodyResponse = withoutBody ? undefined : ((await response.json()) as T);
+  } catch (e) {
+    bodyResponse = {};
+  }
+
+  const errorMessageDefault = response.statusText || 'Unknown error';
+
+  throw new FetchResponseError({
+    code:
+      response.status === 400 || invalidStatus?.includes(response.status)
+        ? 'invalid'
+        : 'exception',
+    status: response.status,
+    message: errorMessage
+      ? errorMessage[response.status]
+        ? errorMessage[response.status]
+        : typeof errorMessage === 'string'
+        ? errorMessage
+        : errorMessageDefault
+      : errorMessageDefault,
```

Comment:
I find this part a bit confusing, with too many ternary operators.
Maybe process the message value sooner, in a more reader friendly way ?